src/main/java/hudson/plugins/git/extensions/impl/SubmoduleOption.java

```diff
@@ -126,6 +137,7 @@ public void onCheckoutCompleted(GitSCM scm, Run<?, ?> build, GitClient git, Task
                         .parentCredentials(parentCredentials)
                         .ref(build.getEnvironment(listener).expand(reference))
                         .timeout(timeout)
+                        .threads(threads)
```

Comment:
Could you validate the set `threads` value (similar to the one for `depth`)?
I suggtest to replace line 140 with something like this between line 146 and 147:
```
int usedThreads = threads == null || threads < 1 ? 1 : threads;
cmd.threads(usedThreads);
```
The `null` check will be required anyway because I suggest to change the method signature for `SubmoduleUpdateCommand::threads` from taking an `Integer` to taking an `int` because `null` values are not required in there.